src/matchbox/matchbox.py

```diff
@@ -199,6 +200,8 @@ def not_matching(self, value):
         :returns: set of objects that doesn't match characteristic value
         :rtype: set
         """
+        if value is None:
```

Comment:
not needed actually. If this behaviour will be desired, index should rather be skipped instead from checking manually, rather than arbitrary skipped by MatchBox itself.